### PR TITLE
CPDP-1374

### PR DESCRIPTION
**Pull request by sam-miele**

### CODE REVIEW

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/exemptionandlosses/annual_exempt_amount.scala.html [hunk 1]
```diff
@@ -76,10 +77,16 @@
...
+        } else {
+          <p>@Html(messages(s"$key.link", appConfig.annualExemptAmountUrl))</p>
+        }
 
         @submitButton(messages("button.saveAndContinue"))
     }
 
     @returnToSummaryLink(displayReturnToSummaryLink = true)
 }
+
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/exemptionandlosses/annual_exempt_amount.scala.html]: the help text needs to change for trusts as well

File: conf/application.conf [hunk 2]
```diff
@@ -191,6 +191,8 @@ external-url {
   reliefs-info = "https://www.gov.uk/hmrc-internal-manuals/capital-gains-manual/cg60201c"
   letting-relief-info = "https://www.gov.uk/tax-sell-home/let-out-part-of-home"
   market-value = "https://www.gov.uk/capital-gains-tax/market-value"
+  tax-free-allowance="https://www.gov.uk/guidance/trusts-and-capital-gains-tax#tax-free-allowance"
```
Comment (AndyHWChung) [inline conf/application.conf]: `trusts-tax-free-allowance` here and in the `val` in the config would be more precise

File: conf/messages [hunk 3]
```diff
@@ -959,14 +959,26 @@ previousYearsLossesValue.error.required=Enter an amount of Capital Gains Tax los
 previousYearsLossesValue.label=Amount of losses from previous tax years
 previousYearsLosses.link=<a href="{0}" target="_blank">Find out how to report losses (opens in a new window)</a>.
 
-annualExemptAmount.title=How much of your Capital Gains Tax Annual Exempt Amount do you want to use?
-annualExemptAmount.helpText=In the tax year {0} to {1}, you only have to pay Capital Gains Tax on gains above {2}.
+annualExemptAmount.individual.title=How much of your Capital Gains Tax Annual Exempt Amount do you want to use?
+annualExemptAmount.trust.title=How much of the trust’s Capital Gains Tax Annual Exempt Amount does it want to use?
+
+annualExemptAmount.individual.helpText=In the tax year {0} to {1}, you only have to pay Capital Gains Tax on gains above {2}.
+annualExemptAmount.trust.helpText=In the tax year 2020 to 2021, most trustees only have to pay Capital Gains Tax on gains above £6,150. This is called the Annual Exempt Amount. <br/> <br/>The Annual Exempt Amount for trusts for vulnerable beneficiaries is £12,300.
```
Comment (AndyHWChung) [inline conf/messages]: the years and the values of the annual exempt amounts should come from the config as parameters

File: conf/messages [hunk 4]
```diff
@@ -959,14 +959,26 @@ previousYearsLossesValue.error.required=Enter an amount of Capital Gains Tax los
...
+
+annualExemptAmount.trust.details.1.header=What is the Annual Exempt Amount?
+annualExemptAmount.trust.details.1.body=Each tax year a trust gets an annual tax-free allowance. This is known as the <a href="{0}"> Annual Exempt Amount (opens in a new window)</a>. Trustees only pay Capital Gains Tax if the trust's taxable gains for the tax year are above the Annual Exempt Amount.
```
Comment (AndyHWChung) [inline conf/messages]: is that a straight quote in `trust's`?

Comment (AndyHWChung) [reply conf/messages]: also we need `target="_blank"`  for links that open in a new window

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/exemptionandlosses/annual_exempt_amount.scala.html (hunk 1 above)
Comment (sam-miele) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/exemptionandlosses/annual_exempt_amount.scala.html]: It does at the moment. Maybe it's not very clear, happy to change it. But the help text message key gets read as `annualExemptAmount.individual.helpText` or annualExemptAmount.trust.helpText` on line 65

File: conf/messages [hunk 5]
```diff
@@ -959,14 +959,26 @@ previousYearsLossesValue.error.required=Enter an amount of Capital Gains Tax los
...
+
+annualExemptAmount.individual.helpText=In the tax year {0} to {1}, you only have to pay Capital Gains Tax on gains above {2}.
+annualExemptAmount.trust.helpText=In the tax year {0} to {1}, most trustees only have to pay Capital Gains Tax on gains above £6,150. This is called the Annual Exempt Amount. <br/> <br/>The Annual Exempt Amount for trusts for vulnerable beneficiaries is £12,300.
+
+annualExemptAmount.trust.details.1.header=What is the Annual Exempt Amount?
+annualExemptAmount.trust.details.1.body=Each tax year a trust gets an annual tax-free allowance. This is known as the <a href="{0}" target="_blank"> Annual Exempt Amount (opens in a new window)</a>. Trustees only pay Capital Gains Tax if the trust’s taxable gains for the tax year are above the Annual Exempt Amount.
+annualExemptAmount.trust.details.2.header=Who are vulnerable beneficiaries?
+annualExemptAmount.trust.details.2.body=<a href="{0}">A vulnerable beneficiary (opens in a new window)</a> could be someone under 18 whose parent has died. It could also be a disabled person who is eligible for certain benefits.
```
Comment (AndyHWChung) [inline conf/messages]: missing `target="_blank"` here

File: conf/messages [hunk 6]
```diff
@@ -959,14 +959,26 @@ previousYearsLossesValue.error.required=Enter an amount of Capital Gains Tax los
 previousYearsLossesValue.label=Amount of losses from previous tax years
 previousYearsLosses.link=<a href="{0}" target="_blank">Find out how to report losses (opens in a new window)</a>.
 
-annualExemptAmount.title=How much of your Capital Gains Tax Annual Exempt Amount do you want to use?
-annualExemptAmount.helpText=In the tax year {0} to {1}, you only have to pay Capital Gains Tax on gains above {2}.
+annualExemptAmount.individual.title=How much of your Capital Gains Tax Annual Exempt Amount do you want to use?
+annualExemptAmount.trust.title=How much of the trust’s Capital Gains Tax Annual Exempt Amount does it want to use?
+
+annualExemptAmount.individual.helpText=In the tax year {0} to {1}, you only have to pay Capital Gains Tax on gains above {2}.
+annualExemptAmount.trust.helpText=In the tax year {0} to {1}, most trustees only have to pay Capital Gains Tax on gains above £6,150. This is called the Annual Exempt Amount. <br/> <br/>The Annual Exempt Amount for trusts for vulnerable beneficiaries is £12,300.
```
Comment (AndyHWChung) [inline conf/messages]: we've still got `£6150` and `£12,300` not being passed in as parameters. You can find these in the `TaxYear` in the disposal date under the annual exempt amounts (see here: https://github.com/hmrc/cgt-property-disposals/blob/master/conf/tax_years.conf#L36)

**Comment by AndyHWChung** [comment]
LGTM

**Pull request merged**